modules/javafx.controls/src/main/java/javafx/scene/control/ListCell.java

```diff
@@ -346,6 +347,37 @@ public void changed(
         return new ListCellSkin<>(this);
     }
 
+    /*
+     * The layoutChildren() method is overridden to address a specific accessibility issue: JDK-8309374
+     * If the Accessibility client application requests FOCUS_ITEM before the focused
+     * ListViewSkin/ListCell is created, then JavaFX would return a null object,
+     * and hence accessibility client application cannot draw the focus rectangle.
+     * In this scenario, JavaFX should notify the accessibility application once the
+     * focused ListCell is created and its layout is completed.
+     */
+    /** {@inheritDoc} */
+    @Override protected void layoutChildren() {
+        super.layoutChildren();
+
+        if (isFocused()) {
+            ListView<T> listView = getListView();
+            if (listView != null) {
+                /*
+                 * The notifyAccessibleAttributeChanged() call is submitted via runLater to defer it until after
+                 * the layout completes, because:
+                 * It is possible that when accessibility client application is processing a FOCUS_ITEM notification,
+                 * it may trigger a layout of focused ListItem, and it ends up generating another FOCUS_ITEM change
+                 * notification from here.
+                 * We observed that this scenario occurs when client application is trying to get the
+                 * the focus item property of a focused ListItem's parent(ListView).
+                 * This scenario is avoided by submitting the call via runLater,
+                 * so that the notification is not sent during any getAttribute() call.
+                 */
+                Platform.runLater(() -> listView.notifyAccessibleAttributeChanged(AccessibleAttribute.FOCUS_ITEM));
```

Comment:
What I'm wondering here: Don't the other cells also suffer from this issue?

Reply:
Yes, the issue can be observed with TreeTableCell or TableCell.
I shall create a ticket for them.